### PR TITLE
ci: Add preview workflow

### PR DESCRIPTION
**Pull request by Wentao-Kuang**

### CODE REVIEW

File: infra/basemaps.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { handler as h } from '@basemaps/lambda-tiler';
+import { ConfigProviderMemory, Config, ConfigBundled } from '@basemaps/config';
+import { fsa } from '@chunkd/fs';
+
+process.env['BASEMAPS_ASSEST_LOCATION'] = process.env['ASSETS_LOCATION'];
+
+(async (): Promise<void> => {
```
Comment (blacha) [inline infra/basemaps.ts:7]: you will need to wait for this inside of `handler` so assign this to a local promise.

Comment (Wentao-Kuang) [reply infra/basemaps.ts:7]: Can we change the handler to async here?

Comment (blacha) [reply infra/basemaps.ts:7]: `configLoader.then(() => h(...))`

File: infra/index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      environment: {
+        AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
+        GIT_VERSION: 'v6.28.1',
+        GIT_HASH: '#' + process.env.STAGE,
+        CONFIG_LOCATION: process.env.CONFIG_LOCATION ?? '',
+        ASSETS_LOCATION: process.env.ASSETS_LOCATION ?? '',
+      },
+      bundling: {
+        nodeModules: ['sharp'],
+        externalModules: ['aws-sdk'],
```
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply infra/index.ts:26]: This seems still not working, I have tried to put it into nodeModules and externalModules.

**Comment by Wentao-Kuang** [comment]
Add assets hash location into bundle file.

**Pull request closed without merging**